ER-651 Automated QA results in QA Web

chrisjensenuk opened the pull request:
- Created a `TemplateRunner` which can be shared between Employer and QA web.  This class converts the Automated QA 'data' to a human readable message.  Further templates for other rules need to be registered in IoC and the runner will use them.

- Created `FieldIdResolver.cs` (which is a wrapper around GetQualifiedFieldId) which resolves Vacancy properties to a string when you don't have an instance of Vacancy.

- Added a little fix to the profanity check so Qualifications are correctly inspected.

dev4valtech commented on src/Shared/Recruit.Shared.Web/RuleTemplates/RuleTemplateRunner.cs:
```diff
@@ -0,0 +1,30 @@
+
+        public string ToText(object data, string fieldName)
+        {
+            if (data == null)
```
Where data is null, may be there is a static message.

chrisjensenuk replied:
I think returning an empty string is OK here.  I'll filter out results with an empty message instead when building the VM (This should never happen btw).

sachpatel commented on src/QA/QA.Web/Mappings/ReviewMapper.cs:
```diff
@@ -4,12 +4,16 @@
 using Esfa.Recruit.Vacancies.Client.Domain.Extensions;
 using Esfa.Recruit.Vacancies.Client.Infrastructure.Client;
 using Humanizer;
+using Microsoft.EntityFrameworkCore.Internal;
```
🔥 Please remove

chrisjensenuk replied:
👍

chrisjensenuk commented on src/QA/QA.Web/Views/Review/Review.cshtml:
```diff
@@ -395,12 +422,28 @@
+            });
+        }
+
+        toggleSubmitButton();
```
We could set the submit button text server side but the JS onload will immediately change it anyway so I don't see the benefit of this extra complication.

`hookupHistory()` is shared by `Review.cshtml` and `ReadonlyReview.cshtml`. The js here is only used on this page. Whether it should go into its own module or not I don't have an opinion on.

chrisjensenuk commented:
- The Rule Message Template is now determined by RuleId and not the Type of data
- Added a RuleId enum
- Renamed RuleTemplates to RuleMessageTemplates

dev4valtech commented on src/QA/QA.Web/Mappings/ReviewMapper.cs:
```diff
@@ -27,13 +27,13 @@ public class ReviewMapper
         private readonly IMinimumWageProvider _wageProvider;
         private readonly Lazy<IList<string>> _qualifications;
-        private readonly IRuleTemplateRunner _ruleTemplateRunner;
+        private readonly IRuleTemplateMessageRunner _ruleTemplateRunner;
```
IRuleMessageTemplateRunner may be, matches MessageTemplate naming convention used else where

dev4valtech commented on src/Shared/Recruit.Shared.Web/RuleTemplates/RuleTemplateMessageRunner.cs:
```diff
@@ -0,0 +1,23 @@
+
+namespace Esfa.Recruit.Shared.Web.RuleTemplates
+{
+    public class RuleTemplateMessageRunner : IRuleTemplateMessageRunner
```
RuleMessageTemplateRunner
